.github/workflows/api-compatibility.yml

```diff
@@ -0,0 +1,45 @@
+name: "Inform Incompatible PRs"
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+
+  Check-Compatibility:
+    runs-on: macos-latest
+    steps:
+
+      - name: Checkout-Main
+        uses: actions/checkout@v2
+        with: 
+          ref: ${{ github.base_ref }}
+          path: ${{ github.base_ref }}
+
+      - name: Checkout-HEAD
+        uses: actions/checkout@v2 
+        with: 
+          path: ${{ github.head_ref }}
+      
+      - name: Install-Tools
+        run: |
+          cd ${{ github.base_ref }}
+          make install-tools 
+
+      - name: Generate-Compare-Check-apidiff
+        run: |
+          echo ::group::Generating apidiff states
+          cd ${{ github.head_ref }}
+          export GOPATH=$HOME/go
+          export PATH=$GOPATH/bin:$PATH
+          make apidiff-build
+          cd ../${{ github.base_ref }}
+          echo ::endgroup::
+          
+          echo ::group::Comparing apidiff states
+          make apidiff-compare-GA input_dir="../${{ github.head_ref }}/internal/data/apidiff"
+          echo ::endgroup::
+
+          echo ::group::Checking incompatible apis
+          make apidiff-check input_dir="../${{ github.head_ref }}/internal/data/apidiff"
+          echo ::endgroup::
```

Comment:
Please add comments to explain how this works. Github's action language is not self-explanatory.

Reply:
I have added comments to the workflow and separated parts of the run. Let me know if anything needs to be further enhanced.